das/daser.go

```diff
@@ -15,28 +15,152 @@ import (
 	"github.com/celestiaorg/celestia-node/share"
 )
 
-var log = logging.Logger("das")
+var (
+	log = logging.Logger("das")
 
-// TODO: parameters needs performance testing on real network to define optimal values
-const (
-	//  samplingRange is the maximum amount of headers processed in one job.
-	samplingRange = 100
+	ErrInvalidOptionValue = func(optionName string, value string) error {
+		errorMsg := fmt.Sprintf("das/daser: invalid option value: %s, cannot be %s", optionName, value)
+		return errors.New(errorMsg)
+	}
+)
+
+// optionSetter:
+//
+//	this is expected to be a closure that encloses the operation of parameter setting along side with
+//	the value to be set.
+//	example: both d and a are in the "global scope" relatively to the closure, thus no parameter passing is required.
+//	   func() {
+//
+// .       d.params.option = a
+//
+//	}
+type optionSetter func()
+type Option func(*DASer) error
+type OptionValidator func(value any, optionSetter optionSetter) error
```

Comment:
TBH, this seems over-engineered though I totally understand where you are coming from. The approach from your [original comment](https://github.com/celestiaorg/celestia-node/pull/1225#issuecomment-1285258589) is much simpler and easier to understand. Basically, it feels to much for simple option setters

Reply:
🤔 I thought about this a while before commiting to this road. It does feel over-engineered, however the reason I went with `OptionValidator` is because I did not want the `WithOption` setter to become an "un-patterned" function full of senseless if elses... But quit frankly at second thought, it is a bit too much. Reverting to the initial proposition.